### PR TITLE
Beautify debugging by using debug_struct for mach

### LINKED ISSUE #24: needs `debug_struct`, etc., in Debug impls
E.g.:

```rust
        impl fmt::Debug for Rel {
            fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
                let sym = r_sym(self.r_info);
                let typ = r_type(self.r_info);
                write!(f,
                       "r_offset: {:x} r_typ: {} r_sym: {}",
                       self.r_offset,
                       typ,
                       sym
                )
            }
        }
```

should use a `debug_struct` so can pretty print, etc.  It just looks crappy now :/

### PR DESCRIPTION
**Pull request by lzutao**
Hopefully this one is enough to to fix #24 .

### CODE REVIEW

File: src/mach/symbols.rs [hunk 1]
```diff
@@ -405,18 +405,11 @@ impl<'a> Symbols<'a> {
...
-            match res {
-                Ok((name, nlist)) => {
-                    writeln!(fmt, "{: >10x} {} sect: {:#x} type: {:#02x} desc: {:#03x}", nlist.n_value, name, nlist.n_sect, nlist.n_type, nlist.n_desc)?;
-                },
-                Err(error) => {
-                    writeln!(fmt, "  Bad symbol, index: {}, sym: {:?}", i, error)?;
-                }
-            }
-        }
-        writeln!(fmt, "}}")
```
Comment (lzutao) [inline src/mach/symbols.rs]: Should we keep this? And how?

Comment (philipc) [reply src/mach/symbols.rs]: The elf symtab [prints them as a vec](https://github.com/m4b/goblin/blob/4c24bc5a615960b99a6eb80b36c34497f99ece96/src/elf/sym.rs#L493). We use this as a lazy way of implementing the `rdr` example, and it's handy for it to print out the symbols too, so I think we should keep it.

Comment (lzutao) [reply src/mach/symbols.rs]: Fixed.

Comment (philipc) [reply src/mach/symbols.rs]: Looking again, we don't actually use this for the `rdr` example (`MachO` creates a vec for its symbols instead), but I think this is still fine.

**Pull request merged**